### ISSUE TITLE
std::vector<Point<dim>> support in HDF5

### ISSUE THREAD

**Issue by dangars** [dealii/dealii#7963]
As we have discussed in https://github.com/dealii/dealii/pull/6747#discussion_r277122834, it would be good to add support for `std::vector<Point<dim>>` in the HDF5 interface.

I think that the best way to save a `std::vector<Point<dim>>` in HDF5 is to consider `Point<dim>` as a compound HDF5 type. The advantages are:
* `std::complex<double>` is already is already a compound type
* All the machinery is already in place. I just have to add the types `Point<1>`, `Point<2>` and `Point<3>` to HDF5::internal::get_hdf5_datatype()
* It will be possible to create 1D, 2D or 3D datasets of `Points`. In the tutorial step-62 I used a 1D dataset of Points. In another tutorial we may want to have a 2D probe, in that case we may need a 2D dataset of Points.
* Python understands quite well HDF5 compound types

I will submit a PR in a couple of days with the support for `Point<dim>`

**Comment by drwells** [comment]
This isn't a high-priority issue and the conversion is easy enough - lets close for now.

**Comment by dangars** [comment]
Hi @drwells, thanks for closing this. I should I have closed it long time ago.

I realized that this could introduce bugs.

HDF5 compound types need to know to position in memory of the variable members. That is easy with `std::complex` because the position is fixed in the standard and that is not going to change.

The problem with `Point<dim>` is that if I write the conversion based on the current state of `Point<dim>` and somebody in the future changes the order of the variable members the code would break. That would be very difficult to debug.

I would keep this closed.
